tests/panda/test_writer.py

```diff
@@ -0,0 +1,181 @@
+    )
+
+    set_sim_value(
+        sim_panda.block2.test_capture, Capture.No  # type: ignore[attr-defined]
+    )
+
+    return sim_panda
+
+
+@pytest.fixture
+async def sim_writer(tmp_path, sim_panda) -> PandaHDFWriter:
+    dir_prov = StaticDirectoryProvider(str(tmp_path), "test")
+    async with DeviceCollector(sim=True):
+        writer = PandaHDFWriter(
+            "TEST-PANDA", dir_prov, lambda: "test-panda", panda_device=sim_panda
+        )
+
+    return writer
+
+
+async def test_get_capture_signals_gets_all_signals(sim_panda):
+    async with DeviceCollector(sim=True):
+        sim_panda.test_seq = Device("seq")
+        sim_panda.test_seq.seq1_capture = SignalR(
+            backend=SimSignalBackend(str, source="seq1_capture")
+        )
+        sim_panda.test_seq.seq2_capture = SignalR(
+            backend=SimSignalBackend(str, source="seq2_capture")
+        )
+        await asyncio.gather(
+            sim_panda.test_seq.connect(),
+            sim_panda.test_seq.seq1_capture.connect(),
+            sim_panda.test_seq.seq2_capture.connect(),
+        )
+    capture_signals = get_capture_signals(sim_panda)
+    expected_signals = [
+        "block1.test_capture",
+        "block2.test_capture",
+        "test_seq.seq1_capture",
+        "test_seq.seq2_capture",
+    ]
+    for signal in expected_signals:
+        assert signal in capture_signals.keys()
+
+
+async def test_get_signals_marked_for_capture(sim_panda):
+
+    capture_signals = {
+        "block1.test_capture": sim_panda.block1.test_capture,
+        "block2.test_capture": sim_panda.block2.test_capture,
+    }
+    signals_marked_for_capture = await get_signals_marked_for_capture(capture_signals)
+    assert len(signals_marked_for_capture) == 1
+    assert (
+        signals_marked_for_capture["block1.test"]["capture_type"] == Capture.MinMaxMean
+    )
+
+
+async def test_open_returns_correct_descriptors(sim_writer: PandaHDFWriter):
+    assert hasattr(sim_writer.panda_device, "data")
+    cap1 = sim_writer.panda_device.block1.test_capture  # type: ignore[attr-defined]
+    cap2 = sim_writer.panda_device.block2.test_capture  # type: ignore[attr-defined]
+    set_sim_value(cap1, Capture.MinMaxMean)
+    set_sim_value(cap2, Capture.Value)
+    description = await sim_writer.open()  # to make capturing status not time out
+    assert len(description) == 4
+    for key, entry in description.items():
+        assert entry.get("shape") == [1]
+        assert entry.get("dtype") == "number"
+        assert isinstance(key, str)
+        assert "source" in entry
+        assert entry.get("external") == "STREAM:"
+    expected_datakeys = [
+        "test-panda.block1.test.Min",
+        "test-panda.block1.test.Max",
+        "test-panda.block1.test.Mean",
+        "test-panda.block2.test.Value",
+    ]
+    for key in expected_datakeys:
+        assert "test-panda.block1.test.Min" in description
```

Comment:
shouldn't this be 
`assert key in description`?

Reply:
yeah, thanks!